src/imgui/Shortcuts.hh

```diff
@@ -0,0 +1,95 @@
+#ifndef IMGUI_SHORTCUTS_HH
+#define IMGUI_SHORTCUTS_HH
+
+#include "ImGuiUtils.hh"
+
+#include "strCat.hh"
+#include "imgui_stdlib.h"
+
+#include <functional>
+#include <optional>
+#include <string_view>
+#include <vector>
+
+namespace openmsx {
+
+class ImGuiManager;
+class GlobalSettings;
+class SettingsConfig;
+
+enum ShortcutIndex {
+	GOTO_MEM = 0,
+	STEP,
+	BREAK,
+	GOTO_DISASM,
+	MOVE_UP_MEM,
+	MOVE_DOWN_MEM,
+	MOVE_LEFT_MEM,
+	MOVE_RIGHT_MEM,
+	NUM_SHORTCUTS,
+
+	INVALID,
+};
+
+enum ShortcutType {
+	LOCAL,
+	GLOBAL,
+};
+
+class Shortcuts
+{
+public:
+	Shortcuts(const Shortcuts&) = delete;
+	Shortcuts& operator=(const Shortcuts&) = delete;
+	explicit Shortcuts(SettingsConfig& config_);
+
+	// Shortcuts
+	struct Data {
+		ImGuiKeyChord keyChord;
+		ShortcutType type;
+		bool repeat = false;
+		bool operator==(const Data& other) const {
+			return keyChord == other.keyChord && type == other.type && repeat == other.repeat;
+		}
```

Comment:
Detail:
This code is correct.

Starting from c++20 you can also write:
```[[nodiscard]] bool operator==(const Data&) const = default;```
and then the compiler will implement this operator for you: it will simply compare all fields. And if later new fields are added it will be automatically updated.